content/en/blog/_posts/2022-12-6-cross-namespace-data-sources-alpha.md

```diff
@@ -0,0 +1,137 @@
+
+### Prerequisites
+
+* Kubernetes cluster is being deployed with `AnyVolumeDataSource` and `CrossNamespaceVolumeDataSource` features enabled
+* There are two namespaces, dev and prod
+* CSI driver is being deployed
+* There is an acquired VolumeSnapShot named of "new-snapshot-demo" on prod
+* ReferenceGrants CRD is being deployed
+
+### Add referencegrants permission to CSI Provisioner RBAC
+
+Access to referencegrants is only needed when the CSI driver
+has the `CrossNamespaceVolumeDataSource` controller capability.
+Therefore, external-provisioner adds "get", "list", "watch"
+permissions for "referencegrants" on "gateway.networking.k8s.io".
+
+```yaml
+  - apiGroups: ["gateway.networking.k8s.io"]
+    resources: ["referencegrants"]
+    verbs: ["get", "list", "watch"]
+```
+
+### The CSI Provisioner controller and enable the CrossNamespaceVolumeDataSource feature gate
+
+Add `--feature-gates=CrossNamespaceVolumeDataSource=true` to csi-provisioner container
+
+```yaml
+      - args:
+        - -v=5
+        - --csi-address=/csi/csi.sock
+        - --feature-gates=Topology=true
+        - --feature-gates=CrossNamespaceVolumeDataSource=true
+        image: csi-provisioner:latest
+        imagePullPolicy: IfNotPresent
+        name: csi-provisioner
+```
+
+### Create a ReferenceGrant. Here's a manifest for an example ReferenceGrant.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferenceGrant
+metadata:
+  name: allow-prod-pvc
+  namespace: prod
+spec:
+  from:
+  - group: ""
+    kind: PersistentVolumeClaim
+    namespace: dev
+  to:
+  - group: snapshot.storage.k8s.io
+    kind: VolumeSnapshot
+    name: new-snapshot-demo
+```
+
+### Create a PersistentVolumeClaim by using cross namespace data source
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: example-pvc
+  namespace: dev
+spec:
+  storageClassName: csi-hostpath-sc
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  dataSourceRef:
+    apiGroup: snapshot.storage.k8s.io
+    kind: VolumeSnapshot
+    name: new-snapshot-demo
+    namespace: prod
+  volumeMode: Filesystem
+```
+
+## How can I learn more?
+
+The enhancement proposal,
+[Provision volumes from cross-namespace snapshots](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3294-provision-volumes-from-cross-namespace-snapshots), includes lots of detail about the history and technical implementation of this feature.
+
+Please get involved by joining the Kubernetes SIG Storage to help us enhance this
+feature. There are a lot of good ideas already and we'd be thrilled to have more!
+
+## Acknowledgment
+
+It takes a wonderful group to make wonderful software.
+Thanks are due to everyone who has contributed to the CrossNamespaceVolumeDataSouce feature,
+especially (in alphabetical order) Tim Hockin, Masaki Kimura, Michelle Au and Xing Yang.
+It’s been a joy to work with y’all on this.
```

Comment:
(nit)
```suggestion
It’s been a joy to work with y'all on this.
```
The static site generator may turn this into a different character at render time.